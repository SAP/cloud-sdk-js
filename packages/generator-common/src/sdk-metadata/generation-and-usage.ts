import { InstructionWithText } from './sdk-metadata-types';

export function getGenerationSteps(
  installCommand: string,
  generateCommand: string,
  generationDocLink: string
): InstructionWithText {
  return {
    instructions: `<ul>
<li>Download the API specification to your local computer.</li>
<li>Install the generator <code>${installCommand}</code></li>
<li>Execute the generator <code>${generateCommand}</code></li>
</ul>
The steps above will generate a basic TypeScript client. For additional options like transpiling to JavaScript etc. visit our <a href="${generationDocLink}">documentation.</a>`,
    text: 'Follow the following generation steps to generate the client.'
  };
}

export const apiSpecificUsageText = 'Find a usage example for this API below.';

export const genericUsageText =
  'Find a generic example on how to execute requests with the SAP Cloud SDK below.';
