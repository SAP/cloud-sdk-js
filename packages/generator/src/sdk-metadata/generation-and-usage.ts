import { MultiLineText } from '@sap-cloud-sdk/generator-common/internal';
import { VdmServiceMetadata } from '../vdm-types';
import {
  actionImportCodeSample,
  entityCodeSample,
  functionImportCodeSample
} from './code-samples';
import { sampleOperationImport, getODataEntity } from './code-sample-util';

/**
 * @internal
 */
export function getApiSpecificUsage(
  service: VdmServiceMetadata
): MultiLineText {
  if (service.entities?.length > 0) {
    const entity = getODataEntity(service.originalFileName, service.entities);
    return entityCodeSample(
      entity.className,
      service.className,
      service.directoryName
    );
  }
  // Return function/action import usage if no entity is found.
  if (service.functionImports?.length > 0) {
    const functionImport = sampleOperationImport(
      service.originalFileName,
      service.functionImports
    );
    return functionImportCodeSample(
      functionImport,
      `${service.directoryName}/function-imports`
    );
  }
  if (service.actionImports) {
    const actionImport = sampleOperationImport(
      service.originalFileName,
      service.actionImports
    );
    return actionImportCodeSample(
      actionImport,
      `${service.directoryName}/action-imports`
    );
  }
  return '';
}
