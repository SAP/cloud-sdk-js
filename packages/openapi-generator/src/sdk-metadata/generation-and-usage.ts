import { MultiLineText } from '@sap-cloud-sdk/generator-common/internal';
import { OpenApiDocument } from '../openapi-types';
import { apiSpecificCodeSample } from './code-sample';
import { getMainApi, getMainOperation } from './code-sample-util';

/**
 * @internal
 */
export function getApiSpecificUsage(
  openApiDocument: OpenApiDocument
): MultiLineText {
  const apisWithOperations = openApiDocument.apis.filter(
    api => api.operations?.length > 0
  );

  if (apisWithOperations.length > 0) {
    const mainApi = getMainApi(openApiDocument.serviceName, apisWithOperations);
    const operation = getMainOperation(mainApi);

    const instructions = apiSpecificCodeSample(
      mainApi.name,
      operation,
      openApiDocument.serviceOptions.directoryName
    );
    return instructions;
  }
  return '';
}
