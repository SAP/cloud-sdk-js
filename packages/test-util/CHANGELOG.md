# @sap-cloud-sdk/test-util

## 2.5.0

### Patch Changes

- Updated dependencies [89f1c423]
  - @sap-cloud-sdk/util@2.5.0
  - @sap-cloud-sdk/connectivity@2.5.0

## 2.4.0

### Patch Changes

- Updated dependencies [04726a35]
- Updated dependencies [0a008674]
- Updated dependencies [0a008674]
- Updated dependencies [0a008674]
  - @sap-cloud-sdk/connectivity@2.4.0
  - @sap-cloud-sdk/util@2.4.0
