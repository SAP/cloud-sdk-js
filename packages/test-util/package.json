{
  "name": "@sap-cloud-sdk/test-util",
  "version": "1.32.1",
  "description": "SAP Cloud SDK for JavaScript test utilities",
  "homepage": "https://community.sap.com/topics/cloud-sdk",
  "license": "Apache-2.0",
  "keywords": [
    "sap-cloud-sdk",
    "cloud-sdk",
    "sap-cloud-platform",
    "test-util"
  ],
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "dist/**/*.js",
    "dist/**/*.js.map",
    "dist/**/*.d.ts",
    "dist/**/*.d.ts.map"
  ],
  "repository": "github:SAP/cloud-sdk",
  "scripts": {
    "compile": "yarn tsc -b",
    "prepare": "yarn compile",
    "test": "yarn jest --coverage --runInBand",
    "test:local": "yarn jest",
    "test:debug": "node --inspect-brk ../../node_modules/.bin/jest --runInBand",
    "check:dependencies": "depcheck ."
  },
  "dependencies": {
    "@sap-cloud-sdk/core": "^1.32.1",
    "@sap-cloud-sdk/util": "^1.32.1"
  }
}
