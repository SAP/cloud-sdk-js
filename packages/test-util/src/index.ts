/**
 * [[include:test-util/README.md]]
 * @packageDocumentation
 * @module @sap-cloud-sdk/test-util
 */

export * from './test-destination-provider';
export * from './test-destination-mocker';
