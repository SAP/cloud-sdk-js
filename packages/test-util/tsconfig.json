{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./dist",
    "tsBuildInfoFile": "./dist/.tsbuildinfo",
    "composite": true
  },
  "include": ["src/**/*.ts"],
  "exclude": [
    "dist/**/*",
    "./test/test/**/*",
    "**/*.spec.ts",
    "node_modules/**/*"
  ],
  "references": [{ "path": "../connectivity" }]
}
