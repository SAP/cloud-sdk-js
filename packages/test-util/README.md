<!-- sap-cloud-sdk-logo -->
<!-- This block is inserted by scripts/replace-common-readme.ts. Do not adjust it manually. -->
<a href="https://sap.github.io/cloud-sdk/docs/js/overview"><img src="https://help.sap.com/doc/2324e9c3b28748a4ae2ad08166d77675/1.0/en-US/logo-with-js.svg" alt="SAP Cloud SDK for JavaScript Logo" height="122.92" width="226.773"/></a>
<!-- sap-cloud-sdk-logo-stop -->

# @sap-cloud-sdk/test-util

This package contains utility functions for testing, like loading credentials or creating test destinations.

## Installation

```
$ npm install @sap-cloud-sdk/test-util
```

## Usage

The test-util package makes writing tests for your SAP Business Technology Platform application more convenient.

For example, you can create a mock destination for your tests by using the function `mockTestDestination`.

```
import { mockTestDestination } from '@sap-cloud-sdk/test-util';
mockTestDestination('mockDestinationName');
```

You need to save your system information in local file `systems.json`:

```json
{
  "systems": [
    {
      "alias": "mockDestinationName",
      "uri": "https://www.example.com"
    }
  ]
}
```

and `credentials.json`:

```json
{
  "credentials": [
    {
      "alias": "SYS_001",
      "username": "username",
      "password": "password"
    }
  ]
}
```

<!-- sap-cloud-sdk-common-readme -->
<!-- This block is inserted by scripts/replace-common-readme.ts. Do not adjust it manually. -->
## Support

The recommended way to get in touch with us is to create an issue on [GitHub](https://github.com/SAP/cloud-sdk-js/issues).
Select the issue category **Bug**, **Feature**, or **Question** depending on the nature of your request.
We try to provide fixes, features and answers as soon as possible.

## Contribute

If you would like to contribute to the SAP Cloud SDK, please make yourself familiar with our [contributing guidelines](https://github.com/SAP/cloud-sdk-js/blob/main/CONTRIBUTING.md) and follow the given instructions.

## Links

- [Official support channel](https://github.com/SAP/cloud-sdk-js/issues/new/choose)
- [Github](https://github.com/SAP/cloud-sdk-js)
- [SAP Cloud SDK Documentation portal](https://sap.github.io/cloud-sdk)
  - [SAP Cloud SDK Documentation portal - Overview](https://sap.github.io/cloud-sdk/docs/js/overview)
  - [SAP Cloud SDK Documentation portal - Getting started guide](https://sap.github.io/cloud-sdk/docs/js/getting-started)
  - [SAP Cloud SDK Documentation portal - API documentation](https://sap.github.io/cloud-sdk/api/latest)
  - [SAP Cloud SDK Documentation portal - Release notes](https://sap.github.io/cloud-sdk/docs/js/release-notes)
- [Sample repository](https://github.com/SAP-samples/cloud-sdk-js)

## License

The SAP Cloud SDK is released under the [Apache License Version 2.0.](http://www.apache.org/licenses/)
<!-- sap-cloud-sdk-common-readme-stop -->
