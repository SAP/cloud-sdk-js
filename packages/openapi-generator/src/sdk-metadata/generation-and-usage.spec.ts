import { dummyOpenApiDocument } from '../../test/test-util';
import { getApiSpecificUsage } from './generation-and-usage';

describe('generation-and-usage', () => {
  it('creates api specific usage for entity', () => {
    expect(getApiSpecificUsage(dummyOpenApiDocument)).toMatchSnapshot();
  });
});
