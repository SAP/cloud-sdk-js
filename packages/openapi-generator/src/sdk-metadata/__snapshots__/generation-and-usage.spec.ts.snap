// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generation-and-usage creates api specific usage for entity 1`] = `
"import { DummyApi } from './generated/dummy-package';

const responseData = await DummyApi.DummyFunction(dummyId, undefined, { query_1: 'myQuery1' }).execute({ destinationName: 'myDestinationName' });"
`;
